src/services/jsonSchemaService.ts

```diff
@@ -273,6 +273,9 @@ export class JSONSchemaService implements IJSONSchemaService {
 	}
 
 	public onResourceChange(uri: string): boolean {
+		// always clear this local cache when a resource changes
+		this.cachedSchemaForResource = undefined;
```

Comment:
Perhaps this could be more fine targeted, but I'm not sure it's worth the complexity.